### PR TITLE
Add settings for Codecov

### PR DESCRIPTION
**Pull request by kaakaa**

### CODE REVIEW

File: go.test.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -e
+echo "" > coverage.txt
+
+for d in $(go list ./... | grep -v vendor); do
+    go test -race -coverprofile=profile.out -covermode=atomic $d
```
Comment (Hanzei) [inline go.test.sh]: What does `-race` do?

File: Makefile [hunk 2]
```diff
@@ -31,7 +31,7 @@ clean:
 	rm -rf dist/*
 
 test:
-	go test -cover -v `glide novendor`
+	sh go.test.sh
```
Comment (Hanzei) [inline Makefile]: This seems to complicated for me. Can we not just run `go test -race -coverprofile=profile.out -covermode=atomic ./poll/`

**Comment by Hanzei** [comment]
The integration looks great @kaakaa!

File: go.test.sh (hunk 1 above)
Comment (kaakaa) [inline go.test.sh]: [Data Race Detector \- The Go Programming Language](https://golang.org/doc/articles/race_detector.html)
But we don't need it. I'll remove `-race` option. Thanks.

File: Makefile (hunk 2 above)
Comment (kaakaa) [inline Makefile]: With this script, it becames unnecessary to rewrite scripts related to tests when creating packages other than `poll`, .
But now there is no plan creating other packages and no problem without it.

I'll remove `go.test.sh`.

File: go.test.sh (hunk 1 above)
Comment (Hanzei) [inline go.test.sh]: But nice to know it exists!

**Pull request merged**